### ISSUE TITLE
Xcode Cloud warnings from the included .swift file

### ISSUE THREAD

**Issue by martinmose** [antoniusnaumann/cargo-swift#31]
Hello,

It would be nice if the "generated" `.swift` file could be formatted.

Maybe if that isn't enough it could be nice if it was possible to specific some settings to the Package.swift.
Something like:

```
swiftSettings: [
    .unsafeFlags(["-suppress-warnings"]),
]),
```

I know it is really nice to have 😇 . I just don't like when I get some warnings on my CI 🙏

**Comment by antoniusnaumann** [comment]
That's a good point. 

There are multiple solutions to this problem:
1. Running a formatter in cargo swift 
    - UniFFI already provides the option to format generated bindings using swiftformat, however I would prefer using an apple first party tool for that so I probably do this in cargo swift invoking swift-format (the formatter provided by apple). As this does not come preinstalled, it is probably a good idea to attempt to run this and just show a warning if it fails.
2. Silencing all warnings as you suggested. 
    - This comes at the advantage of covering a wider range of warnings (and realistically those warnings don't really help anyways, since the generated code is not supposed to be changed by hand). I am still unsure about this, maybe silencing warnings by default and providing a --enable-warnings option is the right choice here.

Can you give the warnings you are getting, are they only formatting related?

**Comment by martinmose** [comment]
Glad to here 🥳. 

To address your points:

I concur, that's an excellent suggestion. Are you referring to [SwiftFormat](https://github.com/nicklockwood/SwiftFormat)?

Generally, I'm not in favor of silencing warnings. However, (as you pointed out) given that the generated code isn't intended for manual edits, it's up for an exception.

Perhaps implement a dual approach? Maybe introduce a setting to toggle warnings off. Moreover, if SwiftFormat is detected on the system, the generated code can be automatically formatted. If this were in place, I'd likely begin by setting up SwiftFormat to see which warnings persist and then opt to silence the warnings if necessary.

On my end, the most frequent warning I received was:
`Keyword 'X' does not need to be escaped in argument list`
It seems a formatter could potentially address this.

Additionally, I encountered several instances of:
```
Forming 'UnsafeMutableRawPointer' to a variable of type 'Optional<CheckedContinuation<String, any Error>>'; this is likely incorrect because 'Optional<CheckedContinuation<String, any Error>>' may contain an object reference
```

No other warnings than does two 😇

**Comment by antoniusnaumann** [comment]
> I concur, that's an excellent suggestion. Are you referring to [SwiftFormat](https://github.com/nicklockwood/SwiftFormat)?
This is what UniFFI uses (but I disabled that since it prints an annoying warning that interfers with the progress spinners). 
I would probably use this one: https://github.com/apple/swift-format, but trying out whatever is installed is probably also fine I guess.

> On my end, the most frequent warning I received was: `Keyword 'X' does not need to be escaped in argument list` It seems a formatter could potentially address this.
That one looks more like what a linter with a --fix option would do and I'd avoid doing changes to the generated code that change syntactically relevant stuff.

> Additionally, I encountered several instances of:
> 
> ```
> Forming 'UnsafeMutableRawPointer' to a variable of type 'Optional<CheckedContinuation<String, any Error>>'; this is likely incorrect because 'Optional<CheckedContinuation<String, any Error>>' may contain an object reference
> ```
> 
> No other warnings than does two 😇
After looking at the autogenerated code, I feel confident now that silencing warnings by default (with a flag to enable them) is pretty okay in this case. This is the solution I will implement (dependent on #32)

**Comment by martinmose** [comment]
Make total sense! 
Thanks a lot for your feedback and the hard work 💪. 
I'm looking forward to the disable option ⭐.

**Comment by antoniusnaumann** [comment]
This should be fixed in `0.5.0-beta01` by adding the suggested swiftSettings in Package.swift by default.

I'll add a CLI flag that re-enables them this weekend.

**Comment by martinmose** [comment]
Thanks for the update! The warnings are now gone 🥳! Good job.
